### PR TITLE
[14.0][IMP] sale_financial_risk_info: Absolute and percentage values of re…

### PR DESCRIPTION
**Pull request by ilyasProgrammer**
…maining risk added to partner and Sale Order as computed fields to make it compatible with domain widget.

### CODE REVIEW

**Review by francesco-ooops — APPROVED** [review]
Functional review ok 

image

now it's possible to use these fields for approval tiers, automated actions, etc.

**Review by elvise — APPROVED** [review]
Functional review ok!

**Comment by pedrobaeza** [comment]
Part of this must be in `account_financial_risk`, not in `sale_financial_risk`.

**Review by carlosdauden — CHANGES_REQUESTED** [review]
Extend https://github.com/OCA/credit-control/tree/14.0/sale_financial_risk_info
or create a new module to avoid penalizing performance in installations that do not need it

**Review by francesco-ooops — APPROVED** [review]
functional ok

@carlosdauden @pedrobaeza better now?

**Comment by francesco-ooops** [comment]
@carlosdauden could you take a look? thanks :)

File: sale_financial_risk_info/models/sale.py [hunk 1]
```diff
@@ -9,6 +9,10 @@ class SaleOrder(models.Model):
     _inherit = "sale.order"
 
     risk_info = fields.Html(compute="_compute_risk_info")
+    risk_remaining_value = fields.Monetary(related="partner_id.risk_remaining_value")
+    risk_remaining_percentage = fields.Float(
+        related="partner_id.risk_remaining_percentage"
+    )
```
Comment (carlosdauden) [inline sale_financial_risk_info/models/sale.py:15]: You can use these new fields in _compute_risk_info:
https://github.com/OCA/credit-control/blob/14.0/sale_financial_risk_info/models/sale.py#L26

https://github.com/OCA/credit-control/blob/14.0/sale_financial_risk_info/models/sale.py#L39-L44

**Review by carlosdauden — APPROVED** [review]

File: sale_financial_risk_info/models/sale.py (hunk 1 above)
Comment (ilyasProgrammer) [reply sale_financial_risk_info/models/sale.py:15]: risk_available and risk_remaining_percentage have different formulas.

Comment (carlosdauden) [reply sale_financial_risk_info/models/sale.py:15]: Ok, in that case you can include the risk_remaining_percentage parameter in `format info_pattern.format(`

risk_available and risk_remaining_value have the same operation `record.credit_limit - record.risk_total`

This is just a suggestion, I already approved the PR

**Comment by francesco-ooops** [comment]
@pedrobaeza merge?

File: sale_financial_risk_info/models/res_partner.py [hunk 2]
```diff
@@ -1,10 +1,36 @@
...
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class ResPartner(models.Model):
     _inherit = "res.partner"
 
     risk_percent_warning = fields.Integer(string="Risk percent warning", default=80)
+
+    risk_remaining_value = fields.Monetary(
```
Comment (pedrobaeza) [inline sale_financial_risk_info/models/res_partner.py]: I still think these fields shouldn't go in any `sale_financial_*` module, as they are general values

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: sale_financial_risk_info/models/res_partner.py (hunk 2 above)
Comment (francesco-ooops) [reply sale_financial_risk_info/models/res_partner.py]: @pedrobaeza in which module do you suggest we could create these fields?

Comment (francesco-ooops) [reply sale_financial_risk_info/models/res_partner.py]: @pedrobaeza when you can, if you coul provide directions we'll try to close this PR :)

**Comment by pedrobaeza** [comment]
I think this should go to `account_financial_risk`, as it's general information, not only sales related one. In fact, visually I think they should go on the left part above total risk information. It can be also compacted in one line to show both data.

Not sure if Carlos' fear is that this affects performance, but I don't think so and even he points to a method that do everything in one shot.

**Comment by ilyasProgrammer** [comment]
updated

**Comment by carlosdauden** [comment]
Let's merge and if there is a performance penalty we analyze it

**Comment by francesco-ooops** [comment]
@pedrobaeza go? :)

**Comment by francesco-ooops** [comment]
@pedrobaeza good for merge?

File: account_financial_risk/static/src/scss/from_view.scss [hunk 3]
```diff
@@ -0,0 +1,24 @@
+.o_form_view {
```
Comment (pedrobaeza) [inline account_financial_risk/static/src/scss/from_view.scss]: Can you please avoid this extra CSS? I think using `<div>` and `class="oe_inline"` you can do the same.

Comment (francesco-ooops) [reply account_financial_risk/static/src/scss/from_view.scss]: updated

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**